Strings larger than SSIZE_MAX may not be supported.
In particular, memchr(string, 0, SIZE_MAX) triggers a warning and may fail.

--- src/strdup.c~	2025-07-14 16:50:44.000000000 +0200
+++ src/strdup.c	2025-07-30 19:23:07.879183830 +0200
@@ -32,7 +32,7 @@
 {
   RCUTILS_CAN_RETURN_WITH_ERROR_OF(NULL);
 
-  return rcutils_strndup(str, SIZE_MAX, allocator);
+  return rcutils_strndup(str, SIZE_MAX/2, allocator);
 }
 
 char *
--- src/string_map.c~	2025-07-14 16:50:44.000000000 +0200
+++ src/string_map.c	2025-07-30 19:23:46.535172900 +0200
@@ -174,8 +174,8 @@
     // note that realloc when the pointer is NULL is the same as malloc
     // note also that realloc will shrink the space if needed
 
-    // ensure that reallocate won't overflow SIZE_MAX
-    if (capacity > (SIZE_MAX / sizeof(key_value_pair_t))) {
+    // ensure that reallocate won't overflow SSIZE_MAX
+    if (capacity > (SIZE_MAX/2 / sizeof(key_value_pair_t))) {
       RCUTILS_SET_ERROR_MSG("requested capacity for string_map too large");
       return RCUTILS_RET_BAD_ALLOC;
     }
